src/lib/downloader.ts

```diff
@@ -211,15 +212,23 @@ export const initDownloads = async () => {
 
     if (episode) {
       const bytesTotal = downloadTask.totalBytes ? convertBytesToHumanReadableString(downloadTask.totalBytes) : '---'
+      const podcast = episode.podcast || {}
 
       downloadTaskStates.push({
+        addByRSSPodcastFeedUrl: podcast.addByRSSPodcastFeedUrl || '',
         bytesTotal,
         bytesWritten: '0 KB',
-        episodeId: episode.id,
-        episodeTitle: episode.title,
+        episodeId: episode.id || '',
+        episodeImageUrl: episode.imageUrl || '',
+        episodeMediaUrl: episode.mediaUrl || '',
+        episodePubDate: episode.pubDate || '',
+        episodeTitle: episode.title || '',
         percent: 0,
-        podcastImageUrl: episode.podcast.shrunkImageUrl || episode.podcast.imageUrl,
-        podcastTitle: (episode.podcast && episode.podcast.title) || '',
+        podcastId: podcast.id || '',
+        podcastImageUrl: podcast.shrunkImageUrl || episode.podcast.imageUrl || '',
```

Comment:
Maybe change this to `podcast.shrunkImageUrl || podcast.imageUrl || ''` since you're already checking for `episode.podcast`